sdk-http/src/main/java/io/sphere/sdk/http/UriTemplate.java

```diff
@@ -0,0 +1,95 @@
+package io.sphere.sdk.http;
+
+import java.io.UnsupportedEncodingException;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * This class implements the simple string expansion of the
+ * URI template specification (https://tools.ietf.org/html/rfc6570).
+ */
+public final class UriTemplate extends Base {
+    private final static Pattern PARAMETER_PATTERN = Pattern.compile("\\{(.*)\\}");
+    private final String uriTemplate;
+    private final Set<String> parameterNames;
+
+    private UriTemplate(final String uriTemplate) {
+        this.uriTemplate = uriTemplate;
+        this.parameterNames = extractParameterNames(uriTemplate);
+    }
+
+    private Set<String> extractParameterNames(final String uriTemplate) {
+        final Matcher matcher = PARAMETER_PATTERN.matcher(uriTemplate);
+        final Set<String> result = new HashSet<>();
+
+        while (matcher.find()) {
+            result.add(matcher.group(1));
+        }
+        return result;
+    }
+
+    /**
+     * Returns true if this uri template has the given parameter.
+     *
+     * @param name the parameter name
+     * @return true iff. this uri template has the given parameter
```

Comment:
It's not a typo, it's an acronym for ([if and only if](https://en.wikipedia.org/wiki/If_and_only_if)) 😄